lib/text/text_engine.js

```diff
@@ -396,7 +396,10 @@ shaka.text.TextEngine = class {
       }
 
       captionsMap.get(id).get(startAndEndTime).push(cue);
-      if (id == this.selectedClosedCaptionId_) {
+      // With multiperiod live DASH DAI streams a comma is appended to the
+      // selectedClosedCaptionId string each time a new period appears in
+      // the manifest e.g. "CC1,,," while the id value remains "CC1"
```

Comment:
That is either a bug in the caption parser or your content.  I don't think this should be handled here.

Reply:
Ok found and fixed the root cause